lib/utils/type.cpp

```diff
@@ -0,0 +1,25 @@
+#include "type.hpp"
+#ifdef __GNUG__
```

Comment:
I'm pretty sure clang supports `__cxa_demangle` so you can probably change the `ifdef` here. I think only Visual Studio doesn't.